iron-form.html

```diff
@@ -11,503 +11,459 @@
+          // has a validate() method, otherwise we can't check it.
+          var validatable = /** @type {{validate: (function() : boolean)}} */ (el);
+          if (validatable.validate) {
+            valid = !!validatable.validate() && valid;
           }
-          json[name].push(value);
         }
-      }
+        return valid;
+      },
 
-      // Go through all of the registered custom components.
-      for (var el, i = 0; el = this._customElements[i], i < this._customElements.length; i++) {
-        // If this custom element is inside a custom element that has already
-        // registered to this form, skip it.
-        if (!this._isChildOfRegisteredParent(el, true) && this._useValue(el)) {
-          addSerializedElement(el.name, el.value);
+      /**
+       * Submits the form.
+       */
+      submit: function(event) {
+        // We are not using this form for submission, so always cancel its event.
+        if (event)
+          event.preventDefault();
+
+        // If you've called this before distribution happened, bail out.
+        if (!this._form || !this.validate()) {
+          return;
         }
-      }
 
-      // Also go through the form's native elements.
-      for (var el, i = 0; el = this.elements[i], i < this.elements.length; i++) {
-        // If this native element is inside a custom element that has already
-        // registered to this form, skip it.
-        if (this._isChildOfRegisteredParent(el, true) || !this._useValue(el)) {
-          continue;
+        // Remove any existing children in the submission form (from a previous submit).
+        while (this.$.helper.children.length !== 0) {
+          this.$.helper.removeChild(this.$.helper.children[0]);
         }
 
-        // A <select multiple> has an array of values.
-        if (el.tagName.toLowerCase() === 'select' && el.multiple) {
-          for (var o = 0; o < el.options.length; o++) {
-            if (el.options[o].selected) {
-              addSerializedElement(el.name, el.options[o].value);
-            }
+        var json = this.serializeForm();
+
+        // If we want a redirect, submit the form natively.
+        if (this.allowRedirect) {
+          // If we're submitting the form natively, then create a hidden element for
+          // each of the values.
+          for (element in json) {
+            this.$.helper.appendChild(this._createHiddenElement(element, json[element]));
           }
+          this.$.helper.submit();
+          this.fire('iron-form-submit');
         } else {
-          addSerializedElement(el.name, el.value);
+          this._makeAjaxRequest(json);
         }
-      }
+      },
 
-      return json;
-    },
+      /**
+       * Resets the form to the default values.
+       */
+      reset: function(event) {
+        // We are not using this form for submission, so always cancel its event.
+        if (event)
+          event.preventDefault();
```

Comment:
single line per whose style guide?! (curious and also alarmed. do i live in the past?)